Kernel/Graphics/Intel/DisplayConnectorGroup.cpp

```diff
@@ -125,20 +125,30 @@ Optional<IntelGraphics::PLLSettings> IntelDisplayConnectorGroup::create_pll_sett
 
 NonnullLockRefPtr<IntelDisplayConnectorGroup> IntelDisplayConnectorGroup::must_create(Badge<IntelNativeGraphicsAdapter>, IntelGraphics::Generation generation, MMIORegion const& first_region, MMIORegion const& second_region)
 {
+    OwnPtr<IntelGraphics::AddressSpace> address_space;
+    {
+        auto paddr = first_region.pci_bar_paddr;
+        // Note: Half the GTTMADDR (PCI BAR 0) is dedicated to MMIO register space, the second half to Virtual memory PTEs, etc.
+        auto address_space_length = first_region.pci_bar_space_length / 2;
+        auto address_space_mapping = MUST(Memory::map_typed<volatile u32>(paddr.offset(address_space_length), address_space_length, Memory::Region::Access::ReadWrite));
+        address_space = MUST(IntelGraphics::AddressSpace::create(generation, move(address_space_mapping)));
```

Comment:
if it's really truly required that this address space exist, then we can play games like this with an immediately invoked lambda expression:

```
 NonnullOwnPtr<IntelGraphics::AddressSpace> address_space = [&]()
    {
        auto paddr = first_region.pci_bar_paddr;
        // Note: Half the GTTMADDR (PCI BAR 0) is dedicated to MMIO register space, the second half to Virtual memory PTEs, etc.
        auto address_space_length = first_region.pci_bar_space_length / 2;
        auto address_space_mapping = MUST(Memory::map_typed<volatile u32>(paddr.offset(address_space_length), address_space_length, Memory::Region::Access::ReadWrite));
        return MUST(IntelGraphics::AddressSpace::create(generation, move(address_space_mapping)));
     }();
```